docs/docs/pyproject.md

```diff
@@ -173,6 +173,26 @@ include = ["CHANGELOG.md"]
 exclude = ["my_package/excluded.py"]
 ```
 
+## dev-only
+
+If set to `true`, this option entirely disables installing runtime dependencies in the development environment.
+This is useful when you do not care about whether it's possible to install the package under development in the development environment, and you don't want to mandate command-line flags to avoid attempting to do so.
```

Comment:
This phrase might read a bit convoluted and proably can be dropped. To me and overpointng a little, the information ist conveys sounds a little like: "This is useful, if you find it useful".

Reply:
Maybe line 179 could be: This is useful when testing your application or library (maybe in its own container environment).